synapse/tests/test_types.py

```diff
@@ -141,6 +141,24 @@ def test_datatype_inet_ipv6(self):
         self.eq( tlib.getTypeParse('inet:srv6', '[AF:00::02]:80'), '[af::2]:80')
         self.eq( tlib.getTypeNorm('inet:srv6', '[AF:00::02]:80'), '[af::2]:80')
 
+    def test_datatype_int(self):
+        tlib = s_types.TypeLib()
+
+        self.assertRaises(BadTypeValu, tlib.getTypeNorm, 'int', 'hehe' )
+        self.assertRaises(BadTypeValu, tlib.getTypeParse, 'int', 'hehe' )
+
+        self.eq( tlib.getTypeNorm('int',11), 11 )
+        self.eq( tlib.getTypeParse('int','11'), 11 )
+
+    def test_datatype_str(self):
+        tlib = s_types.TypeLib()
+
+        self.assertRaises(BadTypeValu, tlib.getTypeNorm, 'str', 11 )
+        self.eq( tlib.getTypeParse('str',11), '11' )
```

Comment:
getTypeParse should not handle anything except strings ( dont want *too* much implicit DWIM perl like behavior ;) )